astropy/cosmology/core.py

```diff
@@ -766,8 +780,8 @@ def nu_relative_density(self, z):
         p = 1.83
         invp = 1.0 / p
         if np.isscalar(z):
-            curr_nu_y = self._nu_y / (1.0 + z)  # only includes massive ones
-            rel_mass_per = (1.0 + (0.3173 * curr_nu_y) ** p) ** invp
+            k = 0.3173 / (1.0 + z)
```

Comment:
There's no derivation, as such.  It's just a numeric fit to a much more complicated expression (which you can derive, but which is much too expensive to evaluate).